### PR TITLE
Include helm chart option for installing jaeger specific services

### PR DESCRIPTION
**Pull request by objectiser**
The recent PR #5656 switched the backend from using zipkin to use jaeger, but only exposed the zipkin service for use by the internal Istio components.

This PR provides an option `zipkin.jaeger.enabled` (which will hopefully soon change chart name so `tracing.jaeger.enabled`) to enable the user to also install the jaeger specific services, which would allow a jaeger tracer (used from within services) to report tracing data in the jaeger native format.

There are three other changes:
1) Renaming app labels from zipkin to jaeger, as this reflects the use of jaeger and doesn't impact any usage of the zipkin service by Istio components
2) Use of the `istio-tracing` name, instead of `zipkin` - thought it would bring it in line with the other istio components - but am happy to change if not suitable - although think it should be `jaeger` rather than `zipkin` (apart from the service).
3) The `service-jaeger.yaml` currently contains a `tracing` service which provides access to the UI. This was just to demonstrate the concept - ideally this could be moved out of the conditional `service-jaeger.yaml` and used as the way to provide access to the tracing UI - rather than it being exposed as a port on the `zipkin` service.

Example of trace from two services that use the jaeger native format for reporting:

istio-0 8-jaeger-services

Example dashboard with the new `istio-tracing` names:

istio-0 8-dashboard

cc @douglas-reid @rshriram

### CODE REVIEW

**Comment by sdake** [comment]
If you want to change the name of the chart, please do so now.  Chart name changes have an upgrade impact, and 0.8 is the first version where these addons are available in the helm template in working fashion.

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -390,6 +390,8 @@ servicegraph:
 
 zipkin:
   enabled: false
+  jaeger:
```
[the remark this answers is not in the document]
Comment (sdake) [reply install/kubernetes/helm/istio/values.yaml]: looks like that PR was manually merged - perhaps the TODO was forgotten.

**Comment by douglas-reid** [comment]
@objectiser in general, i'm fine with this approach. I think `istio-tracing` is a better name. The reason we didn't do this in #5656 was a concern around destabilizing the system.

With this change, do the proxy args need to be updated at all?  Don't we still have args like: `--zipkinAddress zipkin.istio-system:9411` being passed to the proxies (and hardcoded in places like: https://github.com/istio/istio/blob/master/pilot/docker/envoy_policy.yaml.tmpl#L182) ?

**Comment by objectiser** [comment]
@douglas-reid I don't think the proxy args need to change. The format being reported is zipkin, so the service endpoint should reflect that. If later other formats are supported, it then makes it more explicit.

Any thoughts on the other two questions: (1) is it ok to move the `tracing` service out to provide a separate UI service? and (2) how best to enable the additional jaeger specific services, using the current bool `tracing.jaeger.enabled`, as a value property e.g. `tracing.provider = jaeger`, or just enable the jaeger services by default?

**Comment by douglas-reid** [comment]
@objectiser ah, I looked too quickly. I thought you were replacing the `zipkin` service, but rather we are adding new services and changing the app labels on the `zipkin` service.

I don't have any objections to making a new service explicitly for UI concerns. We went halfway on that in the previous PR.

First blush, I like the idea of `tracing.provider = jaeger` options, but I'm fine with using the existing bool for now and fleshing out the approach when the rest of the tracing config story becomes clearer.

**Comment by objectiser** [comment]
@douglas-reid Ok thanks -updates applied, and have left the `tracing.jaeger.enabled` bool for now.

File: install/kubernetes/helm/istio/charts/tracing/templates/deployment.yaml [hunk 2]
```diff
@@ -13,17 +13,23 @@ spec:
...
           imagePullPolicy: {{ .Values.global.imagePullPolicy }}
           ports:
             - containerPort: {{ .Values.service.internalPort }}
             - containerPort: {{ .Values.service.uiPort }}
+            - containerPort: 5775
+              protocol: UDP
+            - containerPort: 6831
+              protocol: UDP
+            - containerPort: 6832
+              protocol: UDP
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/tracing/templates/deployment.yaml]: Who is consuming these?

File: install/kubernetes/helm/istio/charts/tracing/templates/service.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+apiVersion: v1
+kind: List
+items:
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: zipkin
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/tracing/templates/service.yaml]: We don’t need this. We can fix our code to point to istio-tracing

**Comment by rshriram** [comment]
My take: Use jaeger as default. For 0.8 only, add an alias zipkin service that points to the jaeger pod. 

Fix the bootstrap.yaml.tmpl to point to istio-tracing instead of zipkin. Ignore the current args —zipkin cluster etc. they will be removed by 0.9. 

Also fix the global mesh config api to change the name of zipkinService mesh global variable to istio-tracing.

**Comment by douglas-reid** [comment]
@rshriram i think in general we are all in agreement. i'm leaning towards only attempting the follow-on mods of the various bootstraps and static envoy configs and mesh configs to immediately post-0.8. it sounds like that is also what you are suggesting? Or are you suggesting we be more aggressive in 0.8?

**Comment by objectiser** [comment]
@douglas-reid @rshriram  Would be good to handle the config changes in a separate PR anyway - whether for inclusion in 0.8 or post 0.8. So I think the PR is ready to go -but let me know if other changes are required?

@sdake Just checking you are now ok with the PR?

**Comment by objectiser** [comment]
@sdake Thanks. Would you be able to give this the `/lgtm` to trigger the merge?

**Comment by rshriram** [comment]
We could do the config changes in some 0.8.x before final lts release

**Pull request merged**